samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,215 @@
+3. The model is **trained** using the train dataset using LightGbm Lambdarank algorithm.  
+4. The model is **tested** using the test dataset.  This results in a **prediction** that includes a **score** for each hotel instance.  The score is used to determine the ranking relative to other hotels within the same query (e.g. group).  The predictions are then **evaluated** by examining metrics; specifically the [Discounted Cumulative Gain](https://en.wikipedia.org/wiki/Discounted_cumulative_gain).
+5. The final step is to **consume** the model to perform ranking predictions for new incoming hotel searches.
+
+### 1. Label Data
+To label the data with ideal ranking values, the sample follows [Expedia's evaluation guidance](https://www.kaggle.com/c/expedia-personalized-sort/overview/evaluation):
+
+* 0 - The user neither clicked on this hotel nor purchased\booked a room at this hotel.
+* 1 - The user clicked through to see more information on this hotel.
+* 2 - The user purchased\booked a room at this hotel.
+
+Expedia's dataset includes both **Click_Bool** and **Booking_Bool** columns that indicate whether the user has clicked or purchased/booked a hotel.  Applying the above guidelines to these columns, we create a new **Label** column that contains values {0, 1, 2} for each hotel search result.
+
+The code for labeling the data is similar to the following:
+
+```CSharp
+// Load dataset using TextLoader by specifying the type name that holds the data's schema to be mapped with datasets.
+IDataView data = mlContext.Data.LoadFromTextFile<HotelData>(originalDatasetPath, separatorChar: ',', hasHeader: true);
+
+// Create an Estimator and use a custom mapper to transform label hotel instances to values 0, 1, or 2.
+IEstimator<ITransformer> dataPipeline = mlContext.Transforms.CustomMapping(Mapper.GetLabelMapper(mlContext, data), null);
+
+// To transform the data, call the Fit() method.
+ITransformer dataTransformer = dataPipeline.Fit(data);
+IDataView labeledData = dataTransformer.Transform(data);
+
+[...]
+
+// Custom mapper used to label a hotel search result with the ideal rank. 
+public static Action<HotelData, HotelRelevance> GetLabelMapper(MLContext mlContext, IDataView data)
+{
+    Action<HotelData, HotelRelevance> mapper = (input, output) =>
+    {
+        if (input.Srch_Result_Booked == 1)
+        {
+            output.Label = 2;
+        }
+        else if (input.Srch_Result_Clicked == 1)
+        {
+            output.Label = 1;
+        }
+        else
+        {
+            output.Label = 0;
+        }
+    };
+
+    return mapper;
+}
+`````
+### 2. Split Data
+ With the data properly labeled, it is ready to be split into the train/test datasets.  When splitting the data, it's important to make sure that the hotel search results for a given query aren't split across the two datasets.  This would cause label leakage where the same query in our training dataset also exists within the testing dataset.
```

Comment:
Word smithing...
```suggestion
 With the data properly labeled, it is ready to be split into the train/test datasets.  When splitting the dataset, it's 
 important to make sure all of the results for a single hotel search remain in the same dataset split. This would cause data leakage where the same query in our training dataset also exists within the testing dataset.
```